### ISSUE TITLE
Cypress tests are failing with current master

### ISSUE THREAD

**Issue by max-nextcloud** [nextcloud/collectives#1072]
**Describe the bug**
Cypress tests have been failing recently. In particular three tests in pages-links.spec.js:
* Opens link with absolute path to text file in Nextcloud in viewer
* Opens link with relative path to pdf in Nextcloud in viewer 
* Opens link with absolute path to image in Nextcloud in viewer

**To Reproduce**
1. Visit a recent dependency PR such as https://github.com/nextcloud/collectives/pull/1071
2. Check the failed cypress tests on master.
3. Find the failure there

**Expected behavior**
Cypress tests should pass and dependency PRs should be merge automatically.

**Screenshots**

![cypress-screenshot1](https://github.com/nextcloud/collectives/assets/97337118/ebdc97ca-e331-44a5-a6f8-42d871209fdf)
![cypress-screenshot2](https://github.com/nextcloud/collectives/assets/97337118/5464d8d0-e378-4dd4-89b5-332db4a18569)
![cypress-screenshot3](https://github.com/nextcloud/collectives/assets/97337118/b49408e3-7d29-411f-9564-d848f36c083a)

**Server details:**
- Current Cypress master CI

**Client details:**
 - electron

#### Nextcloud log (data/nextcloud.log)

Nextcloud logs are empty (i.e. no errors) for the run in question.

**Comment by max-nextcloud** [comment]
I tried to reproduce this locally with latest master, viewer and collectives but failed.

**Comment by max-nextcloud** [comment]
I tried to bisect which commit in text introduced this but failed. Now that I got the test to fail i cannot get it to pass again. It even failed when i had the same text commit checked out as during the last passing CI run.

Since I had a local failure now i could debug it some. Looks like it the first failing test tries to open `//test.png`:
![grafik](https://github.com/nextcloud/collectives/assets/97337118/8c8eb874-aff5-4b85-b7a8-148fa951d9d5)

But the list of files after fetching the metadata only contains a file called `/test.png` so when the viewer tries to calculate the current index that is `-1` and later the file to open is empty leading to exceptions.

@mejo- does that sound familiar to you?
